keras_nlp/samplers/beam_sampler.py

```diff
@@ -149,7 +145,7 @@ def body(prompt, state, index, log_probs):
             # Compute the softmax distribution for the next token.
             logits, state = next(prompt, state, index)
             vocab_size = tf.shape(logits)[-1]
-            probs = keras.activations.softmax(logits)
+            probs = keras.activations.softmax(logits / self.temperature)
```

Comment:
We will also need to do this for the new constrastive_sampler. (after a merge with master)

Reply:
Made the required changes from above, do let me know if any more are needed.
Also, I would like to take up the task of the `new constrastive_sampler` if possible. Thanks!